lib/application.js

```diff
@@ -102,6 +103,9 @@ module.exports = class Application extends Emitter {
    */
 
   use(fn) {
+    if (this.forceAsync && fn.constructor.name !== 'AsyncFunction') {
+      throw new TypeError('middleware must be an AsyncFunction!');
+    }
```

Comment:
Fair enough, but maybe add a check so you don't get `cannot read property "constructor" of undefined`